osu.Game/Beatmaps/BeatmapOnlineLookupQueue.cs

```diff
@@ -71,40 +79,38 @@ private void lookup(BeatmapSetInfo set, BeatmapInfo beatmapInfo)
 
             var req = new GetBeatmapRequest(beatmapInfo);
 
-            req.Failure += fail;
-
             try
             {
                 // intentionally blocking to limit web request concurrency
                 api.Perform(req);
 
+                if (req.CompletionState == APIRequestCompletionState.Failed)
+                {
+                    logForModel(set, $"Online retrieval failed for {beatmapInfo}");
+                    beatmapInfo.OnlineID = -1;
+                }
```

Comment:
This looks as if there's a missing early return? Is there a need to check `Response` at all if the state is failed?

Reply:
Arguably the response would always be null in this case, but I've changed it to return for safety.